app/views/layouts/_navbar.html.erb

```diff
@@ -0,0 +1,89 @@
+                <%= link_to community.name, community_path(community), :class=>"dropdown-item" %>
+              <% end %>
+            <% end %>
+          </div>
+        </li>
+      </ul>
+      <ul class="navbar-nav">
+        <li class="nav-item dropdown">
+          <%= link_to "Search Community", root_path, :class => "nav-link dropdown-toggle", :id=>"navbarDropdown", "data-toggle"=>"dropdown" %>
+          <div class="dropdown-menu" id="myDropdown" aria-labelledby="navbarDropdown">
+            <input type="text" placeholder="Search.." id="myInput" onkeyup="filterFunction()">
+            <% if @communities.present? %>
+              <% @communities.each do |community| %>
+                <%= link_to community.name, community_path(community), :class=>"dropdown-item" %>
+              <% end %>
+            <% end %>
+          </div>
+        </li>
+      </ul>
+      <form class="form-inline my-2 my-lg-0">
+        <input class="form-control mr-sm-2" id="myDropdown" type="search" placeholder="Search" aria-label="Search">
+      </form>
+      <ul class="navbar-nav">
+        <% if account_signed_in? %>
+          <div class="karma float-left mt-2 mr-2">
+            <%= current_account.karma %>
+            <li class="fa fa-star text-warning"></li>
+          </div>
+          <li class="nav-item dropdown">
+            <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+              <%= current_account.username %>
+            </a>
+            <div class="dropdown-menu" aria-labelledby="navbarDropdown">
+              <%= link_to "Profile", profile_path(current_account.username), class: "dropdown-item" %>
+              <%= link_to "Settings", edit_account_registration_path, class: "dropdown-item" %>
+              <%= link_to "Sign Out", destroy_account_session_path, method: :delete, class: "dropdown-item" %>
+            </div>
+          </li>
+        <% else %>
+          <li class="nav-item">
+            <%= link_to "Log In", new_account_session_path, class: "nav-link" %>
+          </li>
+          <li class="nav-item">
+            <%= link_to "Sign Up", new_account_registration_path, class: "nav-link" %>
+          </li>
+        <% end %>
+      </ul>
+    </div>
+  </nav>
+</div>
+<script>
+  function myFunction() {
+    document.getElementById("myDropdown").classList.toggle("show");
+  }
+
+  function filterFunction() {
+    var input, filter, ul, li, a, i;
+    input = document.getElementById("myInput");
+    filter = input.value.toUpperCase();
+    div = document.getElementById("myDropdown");
+    a = div.getElementsByTagName("a");
+    for (i = 0; i < a.length; i++) {
+      txtValue = a[i].textContent || a[i].innerText;
+      if (txtValue.toUpperCase().indexOf(filter) > -1) {
+        a[i].style.display = "";
+      } else {
+        a[i].style.display = "none";
+      }
+    }
+  }
+</script>
```

Comment:
can you move this to a separate js file?

Reply:
Resolved bug - Navbar js is moved to separate file